### PR TITLE
AL-2490: Add workflows for running e2e and unit tests

### PR DESCRIPTION
**Pull request by dario-archilogic**

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    steps:
+      - name: Set up node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16.9'
+          registry-url: https://registry.npmjs.org/
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Set up SSH key
+        run: bash .github/workflows/setup-ssh.sh ${{ secrets.SSH_KEY }}
```
Comment (bnjm) [inline .github/workflows/e2e-tests.yml:42]: Could we use a Github PAT tied to the archilogic-ci user for this?

**Review by mredele — APPROVED** [review]
Thanks, LGTM!

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (dario-archilogic) [reply .github/workflows/e2e-tests.yml:42]: Think we should be able to. Note that when the reusable workflow is executed, it uses the local GH PAT token generated by the workflow in the project where it is called from (i.e. `${{ github.token }}`).

**Review by bnjm — APPROVED** [review]

**Comment by dario-archilogic** [comment]
@mredele can you pls merge this PR, only authorized users can merge it.

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (bnjm) [reply .github/workflows/e2e-tests.yml:42]: That makes sense, though just to clarify I meant *instead of* this SSH key

Comment (dario-archilogic) [reply .github/workflows/e2e-tests.yml:42]: Just to confirm, we would then pull private repositories like this via `npm`: https://stackoverflow.com/a/28729646

Comment (dario-archilogic) [reply .github/workflows/e2e-tests.yml:42]: Created a ticket to look into this: https://linear.app/archilogic/issue/AL-2781/switch-to-using-github-pat-instead-of-ssh-keys-for-private-npm

**Pull request merged**